x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -76,6 +76,10 @@ public RealmRef getLookedUpBy() {
         return lookedUpBy;
     }
 
+    public RealmRef getNominalRealm() {
```

Comment:
```suggestion
    public RealmRef getRealm() {
```

Reply:
I think `getRealm` is a bit too general - I worry about someone just deciding to call `getRealm()` without thinking about the fact that there's 2 possible realms that might be used and they need to be intentional about which one to use.

Maybe `getSourceRealm()`?

The better long term fix would actually be to make `getLookedUpBy()` always return a realm (that is, is uses this implementation) and then have a `isUserLookup()` if you need to know whether there was a specific (separate) lookup realm.

Reply:
I agree getRealm is too general. I am OK with `getSourceRealm` and will go with it.

Is `User#isRunAs` functionally equivelent to the `isUserLookup` method you propose?

Reply:
Logically equivalent yet, but it's implemented by looking at different fields, so I'd rather have both implementations so that the one in `Authentication` is specifically about whether `lookupRealm` is populated.